django/db/models/sql/query.py

```diff
@@ -1137,7 +1137,29 @@ def build_filter(self, filter_expr, branch_negated=False, current_negated=False,
 
         # Work out the lookup type and remove it from the end of 'parts',
         # if necessary.
-        value, lookups, used_joins = self.prepare_lookup_value(value, lookups, can_reuse, allow_joins)
+        if isinstance(value, (list, tuple)) and value:
+            klass = type(value)
+            new_value = []
+            new_lookups = []
+            new_used_joins = []
```

Comment:
I tried making new_lookups a set but this caused non-deterministic failures when running the tests under postgres, due to the lookups being interpreted as transforms when they're returned out of order.